### PR TITLE
feat(api): list connected users endpoint and docs

### PR DESCRIPTION
**Pull request by RamilGaripov**
refs. metriport/metriport#459

### Description

- Added an endpoint at `user/users` to retrieve a list of all users for a client with their appId, MetriportId, and connectedProviders (see: [this docs file](https://github.com/metriport/metriport/compare/459-list-connected-users?expand=1#diff-cdf12db18f3cc888d50f437ad5184449522f1c5ae1d1ad575c0afca11985ed29))
- Added supporting docs.

### CODE REVIEW

File: api/app/src/routes/user.ts [hunk 1]
```diff
@@ -40,6 +44,40 @@ router.get(
...
 
+/**
+ * GET /user/users
+ *
+ * Gets all users and their connected providers for the specified customer.
+ *
+ * @return { UserIdsAndProviders[] }
+ */
+router.get(
+  "/users",
```
Comment (Goncharo) [inline api/app/src/routes/user.ts:55]: to be more "RESTful" this should be just `GET /user`

File: docs/devices-api/api-reference/user/get-users-and-providers.mdx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Get Users and Providers"
+description: "Returns a list of users with their IDs and connected providers."
+api: "GET /user/users"
+---
+
+## Response
+
+<ResponseField name="connectedUsers" type="array" required>
```
Comment (Goncharo) [inline docs/devices-api/api-reference/user/get-users-and-providers.mdx:9]: an array of what?

File: docs/devices-api/api-reference/user/get-users-and-providers.mdx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  The user's connected providers
```
Comment (Goncharo) [inline docs/devices-api/api-reference/user/get-users-and-providers.mdx:10]: this isn't accurate, it's an object containing multiple properties - let's describe the response properly

see here for an example https://github.com/metriport/metriport/blob/4036cfc2f4c19c53142ad29658a25eafe9bea59b/docs/medical-api/api-reference/patient/list-patients.mdx

File: docs/devices-api/api-reference/user/get-users-and-providers.mdx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Get Users and Providers"
```
Comment (Goncharo) [inline docs/devices-api/api-reference/user/get-users-and-providers.mdx:2]: why don't we just call this "Get Connected Users" ?

File: docs/devices-api/api-reference/user/get-users-and-providers.mdx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  ]
+}
+```
+
+<ResponseExample>
+
+```javascript Metriport SDK
+import { MetriportDevicesApi } from "@metriport/api";
+const metriportClient = new MetriportDevicesApi("YOUR_API_KEY");
+const connectedProviders = await metriportClient.getUsersAndProviders();
```
Comment (Goncharo) [inline docs/devices-api/api-reference/user/get-users-and-providers.mdx:34]: `const connectedUsers`

File: docs/devices-api/more-info/models.mdx [hunk 6]
```diff
@@ -208,6 +210,16 @@ export interface User extends MetriportData {
 }
 ```
 
+### User Ids and ProviderSource
+
+```typescript
+export type UserIdsAndProviders = {
```
Comment (Goncharo) [inline docs/devices-api/more-info/models.mdx:216]: We could just call this `ConnectedUser` as this is what it represents?

File: packages/packages/api/src/devices/client/metriport.ts [hunk 7]
```diff
@@ -72,6 +74,16 @@ export class MetriportDevicesApi {
     return resp.data;
   }
 
+  /**
+   * Returns all your users and their connected providers.
+   *
+   * @returns List of users with their IDs and providers.
+   */
+  async getUsersAndProviders(): Promise<{ connectedUsers: UserIdsAndProviders[] }> {
```
Comment (Goncharo) [inline packages/packages/api/src/devices/client/metriport.ts:82]: `getConnectedUsers()`?

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: api/app/src/routes/user.ts [hunk 8]
```diff
@@ -40,6 +44,40 @@ router.get(
...
 
+/**
+ * GET /user/user
+ *
+ * Gets all users and their connected providers for the specified customer.
+ *
+ * @return { ConnectedUserInfo[] }
+ */
+router.get(
+  "/user",
```
Comment (Goncharo) [inline api/app/src/routes/user.ts:55]: to be explicit, we don't want `/user/user` -> just `/user`

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: api/app/src/routes/user.ts (hunk 8 above)
Comment (RamilGaripov) [reply api/app/src/routes/user.ts]: Fixed!

File: api/app/package.json [hunk 9]
```diff
@@ -26,7 +26,7 @@
     "@aws-sdk/client-cloudwatch": "^3.338.0",
     "@medplum/core": "^2.0.18",
     "@medplum/fhirtypes": "^2.0.18",
-    "@metriport/api": "^4.1.7",
+    "@metriport/api": "^4.1.8",
```
Comment (Goncharo) [inline api/app/package.json:29]: why is this getting bumped with a production version - if it's being released into develop?

**Review by Goncharo — COMMENTED** [review]
code LGTM - but question about the api pkg ver

File: api/app/package.json (hunk 9 above)
Comment (Goncharo) [reply api/app/package.json:29]: discussed offline - we'll release to prod to fix this asap

**Review by Goncharo — APPROVED** [review]

**Pull request merged**